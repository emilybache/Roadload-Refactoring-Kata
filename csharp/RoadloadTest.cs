
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using Xunit;

namespace roadload
{
    public class RoadloadTest
    {
        [UseReporter(typeof(DiffReporter))]
        [Fact]
        public void CalculateCO2Values()
        {            
            CombinationApprovals.VerifyAllCombinations(doCalculateCO2Values,
                new bool[]{false, true}, 
                new bool[]{false, true},
                new double?[] {3.5, null},
                new double?[] { 3.2, null },
                new double?[] { 3.1, null }
                );
            
        }

        private static string doCalculateCO2Values(bool isMidPoint, bool isSinglePoint, 
            double? highRoadloadF0, double? midRoadloadF0, double? lowRoadloadF0)
        {
            var values = new CO2Values()
            {
                High_roadload_f0 = highRoadloadF0,
                High_roadload_f1 = 10.6575,
                High_roadload_f2 = 6.7,
                High_roadload_tm = 0.003,
                Mid_roadload_f0 = midRoadloadF0,
                Mid_roadload_f1 = 9.75,
                Mid_roadload_f2 = 5.9,
                Mid_roadload_tm = 0.0025,
                Low_roadload_f0 = lowRoadloadF0,
                Low_roadload_f1 = 8.45,
                Low_roadload_f2 = 5.3,
                Low_roadload_tm = 0.002,
            };

            values.IsMidPoint = isMidPoint;
            values.IsSinglePoint = isSinglePoint;

            CO2Calculator.calculateCO2Values(values);
        
            return $"({values.Low_roadload_combined}, {values.Mid_roadload_combined}, {values.High_roadload_combined})";
        }
    }
}
