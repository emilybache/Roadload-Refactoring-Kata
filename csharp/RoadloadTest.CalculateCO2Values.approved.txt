[False,False,3.5,3.2,3.1] => (0.000422223104721814, , 0.000798757879614663)
[False,False,3.5,3.2,] => (, , 0.000798757879614663)
[False,False,3.5,,3.1] => (0.000422223104721814, , 0.000798757879614663)
[False,False,3.5,,] => (, , 0.000798757879614663)
[False,False,,3.2,3.1] => (0.000422223104721814, , )
[False,False,,3.2,] => (, , )
[False,False,,,3.1] => (0.000422223104721814, , )
[False,False,,,] => (, , )
[False,True,3.5,3.2,3.1] => (, , 0.000798757879614663)
[False,True,3.5,3.2,] => (, , 0.000798757879614663)
[False,True,3.5,,3.1] => (, , 0.000798757879614663)
[False,True,3.5,,] => (, , 0.000798757879614663)
[False,True,,3.2,3.1] => (, , )
[False,True,,3.2,] => (, , )
[False,True,,,3.1] => (, , )
[False,True,,,] => (, , )
[True,False,3.5,3.2,3.1] => (0.000422223104721814, 0.000598534798534799, 0.000798757879614663)
[True,False,3.5,3.2,] => (, 0.000598534798534799, 0.000798757879614663)
[True,False,3.5,,3.1] => (0.000422223104721814, , 0.000798757879614663)
[True,False,3.5,,] => (, , 0.000798757879614663)
[True,False,,3.2,3.1] => (0.000422223104721814, 0.000598534798534799, )
[True,False,,3.2,] => (, 0.000598534798534799, )
[True,False,,,3.1] => (0.000422223104721814, , )
[True,False,,,] => (, , )
[True,True,3.5,3.2,3.1] => (, , 0.000798757879614663)
[True,True,3.5,3.2,] => (, , 0.000798757879614663)
[True,True,3.5,,3.1] => (, , 0.000798757879614663)
[True,True,3.5,,] => (, , 0.000798757879614663)
[True,True,,3.2,3.1] => (, , )
[True,True,,3.2,] => (, , )
[True,True,,,3.1] => (, , )
[True,True,,,] => (, , )
